packages/vite/src/node/server/hmr.ts

```diff
@@ -218,9 +218,18 @@ export function updateModules(
 export async function handleFileAddUnlink(
   file: string,
   server: ViteDevServer,
+  isUnlink: boolean,
 ): Promise<void> {
   const modules = [...(server.moduleGraph.getModulesByFile(file) || [])]
 
+  if (isUnlink) {
+    for (const deletedMod of modules) {
+      server.moduleGraph.idToModuleMap.forEach((mod) => {
+        mod.importers.delete(deletedMod)
+      })
+    }
```

Comment:
Should we traverse each `deletedMod.importedModules` and `importedMod.importers.delete(deletedMod)` instead? That way we don't have to iterate all the modules.

Reply:
Good point!